addons/mass_mailing/models/mailing.py

```diff
@@ -359,6 +359,9 @@ def action_schedule(self):
 
     def action_put_in_queue(self):
         self.write({'state': 'in_queue'})
+        cron = self.env.ref('mass_mailing.ir_cron_mass_mailing_queue')
+        for record in self:
```

Comment:
Just a thought: couldn't we give several "at" so that we could make a create_multi in _trigger (and to cron._trigger(at=records.mapped('schedule_date')) ?

Reply:
I don't like to mixup scalars and containers in the same arguments, it makes it harder to extend (`write` extension ptsd). But What do you think we instead have a separated `_trigger_multi` method ?

    def _trigger(self, at=None):
        at = at or datetime.now()
        return self._trigger_multi(at=[at])
    
    def _trigger_multi(self, at_list):
        ...

Reply:
Indeed mixing argument types is generally a bad idea. Works for me trigger_multi :) .